Robot-Tests/KW_Leveys.robot

```diff
@@ -0,0 +1,251 @@
+
+    Wait Until Element Is Not Visible           ${Spinner_Overlay}
+    Siirry Katselutilaan
+
+
+Leveyden_muokkaus_tallennus  [arguments]                ${testipaikka}
+    Log                                         Testataan leveyden muokkaaminen ja tallennus
+    Testin Aloitus
+    Vaihda Tietolaji                            ${TL_Leveys_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5    20 m
+    Odota sivun latautuminen
+    Siirry Muokkaustilaan
+    Wait Until Element Is Not Visible           ${Map_popup}
+    Click Element At Coordinates                ${Kartta}  0  20
+
+    Wait Until Element Is Visible               ${Leveys_muokattava}
+    Select Radio Button                         road-width    enabled
+    Radio Button Should Be Set To               road-width    enabled
+
+    Wait Until Element Is Visible               ${Leveys_tekstikenttä}
+    Input Text                                  ${Leveys_tekstikenttä}    400    True
+
+    Wait Until Element Is Visible               ${FA_footer_Tallenna}
+    Click Element                               ${FA_footer_Tallenna}
+
+    Wait Until Element Is Not Visible           ${Spinner_Overlay}
+    Siirry Katselutilaan
+    Nollaa Leveys
+
+
+Leveyden katkaisu    [arguments]    ${testipaikka}
+    Log    Katkaisutyökalun testaaminen
+    Vaihda Tietolaji                            ${TL_Leveys_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5    100m
+
+    Odota sivun latautuminen
+    Siirry Muokkaustilaan
+    Log    Leikataan tielinkki kahteen osaan
+    Wait Until Element Is Visible              ${Muokkaustila_Leikkaus}
+    Click Element                              ${Muokkaustila_Leikkaus}
+
+    Odota sivun latautuminen
+    Click Element At Coordinates               ${Kartta}  0  10
+    Wait Until Element Is Visible              ${Leveys_leikkausnäkymä_A}
+    Element Should Be Visible                  ${Leveys_leikkausnäkymä_B}
+
+    Select Radio Button                         road-width-a    disabled
+    Radio Button Should Be Set To               road-width-a    disabled
+    Select Radio Button                         road-width-b    disabled
+    Radio Button Should Be Set To               road-width-b    disabled
+
+    Element Should Be Enabled                   ${FA_footer_Tallenna}
+    Click Element                               ${FA_footer_Tallenna}
+
+    Select Radio Button                         road-width-a    enabled
+    Radio Button Should Be Set To               road-width-a    enabled
+    Select Radio Button                         road-width-b    enabled
+    Radio Button Should Be Set To               road-width-b    enabled
+    
+    Element Should Be Enabled                   ${FA_footer_Peruuta}
+    Click Element                               ${FA_footer_Peruuta}
+
```

Comment:
Miten tää testi toimii, jos se ajetaan useampaan kertaan? Nyt tallennat kantaan muutoksia, mutta alustetaanko tämä nyt jossain uusiksi sitten seuraavaa testikertaa varten?

Muutenkin tässä voisi mielestäni taas jotenkin testata shouldilla, että katkaisun jälkeen pätkillä on eri arvo tms.

Reply:
Muutosta ei tallenneta & lisätty should